nucleus/common/simple-glassfish-api/src/main/java/org/glassfish/embeddable/GlassFishRuntime.java

```diff
@@ -158,38 +162,34 @@ private synchronized static GlassFishRuntime _bootstrap(BootstrapProperties boot
         return me;
     }
 
-    protected synchronized static void shutdownInternal() throws GlassFishException {
-        if (me == null) {
-            throw new GlassFishException("Already shutdown", null);
+    protected void shutdownInternal() throws GlassFishException {
+        synchronized (GlassFishRuntime.class) {
+            if (me == null) {
+                throw new GlassFishException("Already shutdown", null);
+            }
+            me = null;
+            getLogger().info("Completed shutdown of GlassFish runtime");
```

Comment:
It is not new, originally it was printed just on STDOUT and only by one of three "brother" classes.
It is really the last one, because it is in the shutdown hook. It is a part of the "bootstrap" issue, important especially when we restart the server, in some versions was possible situation when one JVM did not completely stop, but another was already starting and both used same files.
When I was starting with this, I reproduced it yet, maybe it was related to hazelcast, now it seems it behaves correctly (Hazelcast doesn't but that will be another issue).